terasoluna-gfw-common-libraries/terasoluna-gfw-common/src/test/java/org/terasoluna/gfw/common/codelist/i18n/SimpleI18nCodeListTest.java

```diff
@@ -21,78 +21,75 @@
 public class SimpleI18nCodeListTest {
     @Autowired
-    @Qualifier("CL_testSetRows01")
-    protected SimpleI18nCodeList testSetRows01;
+    @Qualifier("CL_testSetRows")
+    protected SimpleI18nCodeList testSetRows;
+
+    @Autowired
+    @Qualifier("CL_testSetRowsByCodeList")
+    protected SimpleI18nCodeList testSetRowsByCodeList;
+
+    @Autowired
+    @Qualifier("CL_testSetColumns")
+    protected SimpleI18nCodeList testSetColumns;
+
+    @Autowired
+    @Qualifier("CL_testFallbackTo")
+    protected SimpleI18nCodeList testSetFallbackTo;
 
     @Autowired
-    @Qualifier("CL_testSetRowsByCodeList01")
-    protected SimpleI18nCodeList testSetRowsByCodeList01;
+    @Qualifier("CL_testResolveLocale")
+    protected SimpleI18nCodeList testResolveLocale;
 
     @Autowired
-    @Qualifier("CL_testSetColumns01")
-    protected SimpleI18nCodeList testSetColumns01;
+    @Qualifier("CL_testResolveLocaleFallbackToUS")
+    protected SimpleI18nCodeList testResolveLocaleFallbackToUS;
+
+    @BeforeClass
+    public static void setDefaultLocale() {
+        Locale.setDefault(Locale.US);
+    }
 
     @Test
-    public void testSetRows01() {
-        assertThat(testSetRows01.codeListTable.size(), is(14)); // 2 rows x 7
-                                                                // columns
+    public void testAsMap() {
 
-        Map<String, String> row1 = testSetRows01.asMap(Locale.ENGLISH);
+        Map<String, String> row1 = testSetRows.asMap();
         assertThat(row1, is(notNullValue()));
-        assertThat(row1.keySet().toString(), is("[0, 1, 2, 3, 4, 5, 6]")); // check
-                                                                           // order
+        assertThat(row1.keySet().toString(), is("[0, 1, 2, 3, 4, 5, 6]"));
+
         assertThat(row1.get("0"), is("Sun."));
         assertThat(row1.get("1"), is("Mon."));
         assertThat(row1.get("2"), is("Tue."));
         assertThat(row1.get("3"), is("Wed."));
         assertThat(row1.get("4"), is("Thu."));
         assertThat(row1.get("5"), is("Fri."));
         assertThat(row1.get("6"), is("Sat."));
-
-        Map<String, String> row2 = testSetRows01.asMap(Locale.JAPANESE);
-        assertThat(row2, is(notNullValue()));
-        assertThat(row2.keySet().toString(), is("[0, 1, 2, 3, 4, 5, 6]")); // check
-                                                                           // order
-        assertThat(row2.get("0"), is("日"));
-        assertThat(row2.get("1"), is("月"));
-        assertThat(row2.get("2"), is("火"));
-        assertThat(row2.get("3"), is("水"));
-        assertThat(row2.get("4"), is("木"));
-        assertThat(row2.get("5"), is("金"));
-        assertThat(row2.get("6"), is("土"));
-    }
-
-    @Test(expected = UnsupportedOperationException.class)
-    public void testSetRows02() {
-        // check unmodifiable
-        testSetRows01.asMap(Locale.ENGLISH).put("0", "Sunday");
     }
```

Comment:
- [ ] Tests for `asMap()` must verify that the locale can be resolved using `LocaleContextHolder`.
But I think also you should implement the tests on `AbstractI18nCodeListTest`.

See https://github.com/spring-projects/spring-framework/blob/master/spring-context/src/test/java/org/springframework/context/i18n/LocaleContextHolderTests.java#L67-L68

Reply:
Modified by 6de39c268a32c6c38d619bb742e932e8c592860b

Reply:
The test case `AbstractI18nCodeListTest#testAsMap` is no mean.
And you should verify that System `Locale` and `LocaleContextHolder` do not interfere with each other, ex...

```java
    @Test
    public void testAsMap() {

        AbstractI18nCodeList impl = new AbstractI18nCodeList() {
            @Override
            public Map<String, String> asMap(Locale locale) {
                return Collections.singletonMap("language", locale.getLanguage());
            }
        };

        Locale.setDefault(Locale.ENGLISH);
        assertThat(impl.asMap().get("language"), is(Locale.ENGLISH.getLanguage()));

        LocaleContextHolder.setLocale(Locale.GERMAN);
        assertThat(impl.asMap().get("language"), is(Locale.GERMAN.getLanguage()));

        LocaleContextHolder.setLocale(Locale.FRENCH);
        assertThat(impl.asMap().get("language"), is(Locale.FRENCH.getLanguage()));

        Locale.setDefault(Locale.JAPANESE);
        assertThat(impl.asMap().get("language"), is(Locale.FRENCH.getLanguage()));
    }
```

Reply:
Modified by aedda863dfded9452a0c9a867daf7ad984e1b555
I adopted the example as it is.